src/python/WMCore/WMSpec/StdSpecs/StepChain.py

```diff
@@ -108,6 +115,96 @@ def __call__(self, workloadName, arguments):
             arguments['Step1']['EventsPerLumi'] = self.eventsPerLumi
         return self.workload
 
+    def createStepMappings(self, origArgs):
+        """
+        _createStepMappings_
+
+        Create a simple map of StepName to Step and cmsRun number.
+        cmsRun numbers are sequential, just like the step number.
+        :param origArgs: arguments provided by the user + default spec args
+        :return: update a dictionary in place which is latter used to set a
+         `stepMapping` property in the workload object
+        """
+        for i in range(1, self.stepChain + 1):
+            stepNumber = "Step%d" % i
+            stepName = origArgs[stepNumber]['StepName']
+            cmsRunNumber = "cmsRun%d" % i
+            self.stepMapping.setdefault(stepName, (stepNumber, cmsRunNumber))
+
+    def createStepParentageMappings(self, firstTaskO, origArgs):
+        """
+        _createStepParentageMappings_
+
+        Create a dict struct with a mapping of step name to parent step. It
+        also includes a map of output datasets and parent dataset.
+        :param firstTaskO: a WMTask object with the top level StepChain task
+        :param origArgs: arguments provided by the user + default spec args
+        :return: update a dictionary in place which will be later set as a
+        WMWorkload property
+        """
+        for i in range(1, self.stepChain + 1):
+            stepNumber = "Step%d" % i
+            stepName = origArgs[stepNumber]['StepName']
+            cmsRunNumber = "cmsRun%d" % i
+
+            self.stepParentageMapping.setdefault(stepName, {})
+            self.stepParentageMapping[stepName] = {'StepNumber': stepNumber,
+                                                   'StepCmsRun': cmsRunNumber,
+                                                   'ParentStepName': None,
+                                                   'ParentStepNumber': None,
+                                                   'ParentStepCmsRun': None,
+                                                   'ParentDataset': None,
+                                                   'OutputDatasets': {}}
```

Comment:
Can we change this name to OutputDatasetMap, since top level spec has "OutputDatasets" and wmstatsserver filter will recursively find the that parameter, it will give the mixed results.

Reply:
Sure, I'll update it to `OutputDatasetMap` since it's anyways key'ed by the output module.
Only one detail, in theory this property doesn't go to the request json doc, but it's set/get via WMWorkload object only. If we want to put it in the json doc, then I think we need to filter out some of these things and make a simple {'Step1': [out1, out2], 'Step2': [out3], ...}

Reply:
I think it would be useful to have information in reqmgr couchdb. Even though it won't be used in WMAgent directly. But don't worry about make it simple. We will do that at the time we put the data in reqmgr2 db. I mean we don't have to have the same formation in json and in spec.

Reply:
With that, I think it would be useful to add converter method in the workload helper. basically take current format and convert to whatever simpler format desired. (You don't have to add that in this patch. We can discuss the specific format first and I can add later.) Thanks.